### ISSUE TITLE
Harassment: Followers-Only toots should NOT be visible to nonfollowers mentioned in the toot

### ISSUE THREAD

**Issue by jocelynetienne** [mastodon/mastodon#26937]
### Pitch

FollowersOnly toots should **NOT** be visible to **any** non-follower mentioned in the toot, even if their handle is explicitly mentioned in the toot.

Scenario :
 @badguy@bad.server toots followers-only but mentions @victim@good.server

Upon reception by @good.server of the toot, **if** @victim is not following @badguy@bad.server, the toot is deleted. 
The only persons on @good.server who may see it are @badguy's followers.

### Motivation

Harassers can use Followers-Only toots to facilitate further harassment by letting their crowd know of their post, while keeping the "good" people (victim's followers and other people on victim's server, including moderators) unaware of the harassment posts.

To take an email comparison, this feature would forbid the 
  To:victim BCC:mycrowd type of harassment.

And this can be implemented on the "good server", no need for the compliance of offensive ones (e.g., even if they deactivate this rule).

See thread https://mastodon.gougere.fr/@cyclotopie/111035224050540227

**Comment by ClearlyClaire** [comment]
Hi! Thank you for the suggestion.

I and many people I know rely heavily on followers-only posts, even when mentioning other people, so such a change would unfortunately be a very significant breaking change, and would be very surprising to many users, especially those for whom followers-only is the default posting privacy. Users replying (or just mentioning in a new post) in followers-only would likely assume they are being ignored, and would have no reason to think otherwise.

Furthermore (and this is not to discount different experiences of harassment or other types of abuse), my firsthand experience with abuse and harassment were all posts with public visibility (but which exhibited the same kind of pile-on you describe).

I understand that these harassment posts not being public mean that the report burden is entirely on the victim's, though, and I understand how that is an issue. That being said, I also believe the situation would be very similar with public posts in many situations: replies to an old post, or posts that do not start from a reply, would not be seen by other users in most cases.

Therefore, while I understand this suggestion would provide an immediate stop-gap for certain behavior, I believe it only addresses a small part of it, while breaking long-established legitimate use. (Though the last point could be addressed in large parts with a solid and reliable way to signal that a post was rejected and why it has been rejected—but that is a lot more UX and protocol work).

That being said, we understand the concerns you are raising in this issue, and I personally want to focus on our safety features for end users and moderation tooling for administrators as soon as we are done with releasing 4.2.0 (that is planned for next week); however, that has not been agreed internally yet.

**Comment by trwnh** [comment]
would it make sense to have this be a filter that `victim` can engage, something like the existing "block notifications from people you don't follow" but with an extra condition? this way, `badguy` and `mycrowd` can talk amongst themselves while `victim` sees none of it.

an alternative would be to expand the "block direct messages from people you don't follow" setting to work for all private posts, which would then make it have the behavior you described. (tangentially: that setting needs to be renamed...)

**Comment by ClearlyClaire** [comment]
> would it make sense to have this be a filter that `victim` can engage, something like the existing "block notifications from people you don't follow" but with an extra condition?

Definitely! But that setting is currently problematic since well-meaning people can't expect their notifications to never reach the mentioned people; in addition, those are just about notifications, and the offending replies would still be visible to `victim` whenever they look at their own post in thread views.

**Comment by canuckjacq** [comment]
> would it make sense to have this be a filter that victim can engage, something like the existing "block notifications from people you don't follow" but with an extra condition? this way, badguy and mycrowd can talk amongst themselves while victim sees none of it.

I would love this option. I hate getting replies with this setting. It doesn't make any sense - why only me and their followers? What are they hiding? Why can I see it if I don't follow them?  Really wish we could turn it off. If you don't want to say it publicly, and we're not already buds? Don't say it, or publicly ask me if you can DM me. 

I've seen some groups say it's been a substantial abuse vector, and I've definitely received a few inappropriate replies this way. Now, I'm an admin, so too bad for those dudes, but if it's happening to me, when I'm clearly an admin, it's definitely happening to others.

**Comment by wvholst** [comment]
> 
> Definitely! But that setting is currently problematic since well-meaning people can't expect their notifications to never reach the mentioned people; in addition, those are just about notifications, and the offending replies would still be visible to `victim` whenever they look at their own post in thread views.

How about running the other way: make replies that are follower-only visible to not only the followers of the person replying, but also to all the followers of the person that is being replied to or tagged in the reply? Basically, all the followers of anyone involved in the conversation? Because that would lessen the burden of victims of abusive and harassing replies while still at least technically fulfilling "follower-only".

**Comment by Laconiccrusader** [comment]
This is still an issue in 2025.

**Comment by rvyhvn** [comment]
@wvholst Just wanted to mention that this same visibility issue also affects interactions between **Pixelfed and Mastodon users**.

If a Pixelfed user with a **private account** makes a **followers-only post**, and a Mastodon user (who follows them but isn’t followed back) replies to that post using **followers-only visibility**, the original Pixelfed user never sees the reply — even though they’re directly mentioned.

As others have noted, this happens because “followers-only” content is only visible to the sender’s own followers, not necessarily to those mentioned. This breaks conversations between non-mutuals across federated platforms and causes confusing, invisible replies.

Do you think this specific case (reply visibility across platforms) deserves its own standalone issue?

**Comment by andy-twosticks** [comment]
I **strongly** believe that this is worth a breaking change if it stops one path to harrassment. in this case it's especially nasty because others can't see the abuse.

But with that last point in mind, may I suggest a "less breaking" change that would help?

followers only posts should be public in the poster's profile under "posts and replies".

under normal circumstances we don't look there.  But if you were trying to find out (as a regular user) if another user was abusive, we'd see abusive followers-only posts there. right now we can't.

**Comment by eobet** [comment]
> I and many people

...who are apparently not targets of harassment?

> would it make sense to have this be a filter that victim can engage

...and placing the burden on victims again?

How about for once not forcing vulnerable people to be proactive, but rather:

Add an opt-in option to see followers-only posts mentioning you that's off by default. This would protect everyone while those not facing harassment could flip the options themselves to continue using the feature as before.

For new people, the feature is called "followers only" so logically that doesn't include mentions if the mentioned user isn't a follower (because then the feature should be called "followers and mentions only"). If that's problematic I just want to note that direct messages has a client side warning (that's up to each application to implement) telling the user about the consequences of including mentions. No reason then that this precedence then can't be applied in this instance too.

**Comment by trwnh** [comment]
> the feature is called "followers only" so logically that doesn't include mentions if the mentioned user isn't a follower

then the label is incomplete. it doesn't address *only* your followers. it addresses your followers *and anyone mentioned*. this is because anyone mentioned is *always* addressed; mastodon doesn't have a way to mention someone without addressing them, nor a way to address someone without mentioning them -- the concepts of "mentioning" and "addressing" are collapsed into each other.

> direct messages has a client side warning (that's up to each application to implement) telling the user about the consequences of including mentions. No reason then that this precedence then can't be applied in this instance too.

we could certainly do that, if people are getting confused by the labels. i've argued before that the labels should be more clear about what the option does, but those changes were deemed to be too verbose.

---

> Add an opt-in option to see followers-only posts mentioning you that's off by default

this is what i proposed above:

> expand the "block direct messages from people you don't follow" setting to work for all private posts
